chall04/osalmine.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+
+# Python 3.8.2
+
+import sys
+import fileinput
+
+def append_space(list, size):
+    for i, s in enumerate(list):
+        if len(s) == 0:
+            list[i] = "     "
+        elif len(s) < size:
+            amount = size - len(s)
+            for _ in range(amount):
+                s += ' '
+            list[i] = s
+        elif len(s) > size:
+            raise Exception("line '{0}' shouldn't have more then {1} characters".format(s, size))
+    return list
+
+def replace_sand(list, i, j, size):
+    ind = len(list) - i - 1
+    for k, search in enumerate(list[ind:]):
+        if ind + k is size or list[ind + k + 1][j] is not ' ':
+            tmp = list[ind][:j] + ' ' + list[ind][j + 1:]
+            list[ind] = tmp
+            tmp = list[ind + k][:j] + '.' + list[ind + k][j + 1:]
+            list[ind + k] = tmp
```

Comment:
no need to use tmp here, you can assign right away like
`list[ind] = list[ind][:j] + ' ' + list[ind][j + 1:]`